migrations/V230__Remove_MDM_NCBP_Solution.sql

```diff
@@ -0,0 +1,208 @@
+-- Disable (temp) primary contact and audit triggers to not record below changes
+ALTER TABLE mto_solution
+DISABLE TRIGGER audit_trigger;
+ALTER TABLE mto_common_solution_contact 
+DISABLE TRIGGER trg_ensure_primary_contact_MTO;
+ALTER TABLE mto_common_solution_contact 
+DISABLE TRIGGER audit_trigger;
+ALTER TABLE user_view_customization
+DISABLE TRIGGER audit_trigger;
+
+WITH poc_table AS (
+    SELECT solution.id, "user".common_name, contact.mailbox_title, "user".email, contact.mailbox_address
+    FROM mto_solution AS solution
+    LEFT JOIN mto_common_solution_contact AS contact
+        ON
+            solution.mto_common_solution_key = contact.mto_common_solution_key
+            AND contact.is_primary
+    LEFT JOIN user_account AS "user" ON "user".id = contact.user_id
+    WHERE solution.mto_common_solution_key::TEXT = 'MDM_NCBP' 
+)
+
+-- Replace existing MDM_NCBP from common solution to custom solution
+UPDATE mto_solution AS solution
+SET 
+    mto_common_solution_key = NULL,
+    name = 'Master Data Management - Non-Claims Based Payments',
+    type = 'IT_SYSTEM',
+    modified_by = '00000001-0001-0001-0001-000000000001', -- System Account
+    modified_dts = CURRENT_TIMESTAMP,
+    poc_name = 
+    COALESCE(poc_table.common_name, poc_table.mailbox_title),
+    poc_email = 
+    COALESCE(poc_table.email, poc_table.mailbox_address)
+FROM poc_table
+WHERE
+    solution.id = poc_table.id;
+
+-- Update suggested solution from MDM_NCBP to AMS in milestone solution links
+UPDATE mto_common_milestone_solution_link
+SET 
+    mto_common_solution_key = 'AMS'
+WHERE mto_common_solution_key::TEXT = 'MDM_NCBP';
+
+-- Remove MDM_NCBP from user view customization solutions array
+UPDATE user_view_customization
+SET
+    solutions = ARRAY(
+        SELECT UNNEST(solutions)
```

Comment:
NIT: Would probably be simpler to do array_remove here instead the except trick. Also in general array_remove is better for these kinds of things since the EXCEPT here will dedupe and ruin your order which can sometimes be a surprise, for this enum list I don't think this is an issue though so this is just a nitpick.